.travis.yml

```diff
@@ -1,15 +1,32 @@
-language: node_js
-sudo: false
-node_js: 6
-script:
-- npm run lint
-- npm run test-unit
-before_install:
-- openssl aes-256-cbc -K $encrypted_39958aaf0600_key -iv $encrypted_39958aaf0600_iv
-  -in auth.js.enc -out tests/resources/auth.js -d
-- '[ "${TRAVIS_PULL_REQUEST}" = "false" ] && openssl aes-256-cbc -K $encrypted_39958aaf0600_key
-  -iv $encrypted_39958aaf0600_iv -in auth.js.enc -out tests/resources/auth.js -d ||
-  true'
-- sudo wget https://openwhisk.ng.bluemix.net/cli/go/download/linux/386/wsk -P /usr/local/bin/
-- sudo chmod 711 /usr/local/bin/wsk
-- export PATH=$PATH:"/usr/local/bin/wsk"
+matrix:
+  include:
+    - language: node_js
+      sudo: false
+      node_js: 6
+      script:
+      - npm run lint
+      - npm run test-unit
+      before_install:
+      - 'if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then openssl aes-256-cbc -K $encrypted_39958aaf0600_key -iv $encrypted_39958aaf0600_iv
+        -in auth.js.enc -out tests/resources/auth.js -d; fi'
+      - '[ "${TRAVIS_PULL_REQUEST}" = "false" ] && openssl aes-256-cbc -K $encrypted_39958aaf0600_key
+        -iv $encrypted_39958aaf0600_iv -in auth.js.enc -out tests/resources/auth.js -d ||
+        true'
+      - sudo wget https://openwhisk.ng.bluemix.net/cli/go/download/linux/386/wsk -P /usr/local/bin/
+      - sudo chmod 711 /usr/local/bin/wsk
+      - export PATH=$PATH:"/usr/local/bin/wsk"
+    - language: scala
+      scala:
+        - 2.11.11
+      services:
+        - docker
+      env:
+        global:
+          - TERM=dumb
+      notifications:
+        email: false
+      install: true
+      before_install:
+      - "./tools/travis/setup.sh"
+      script:
+      - "./tools/travis/deploy.sh"
```

Comment:
We should probably model this after the model we are attempting to do now for the other templates/packages, with a structure of 
setup.sh, build.sh, deploy.sh, and test.sh

In your before_install it would be the same as it is now, 
In the script: it would be 
./build.sh && ./deploy.sh && ./test.sh"

Reply:
Thanks eric, just updated these.